### PR TITLE
napi: Fix cpplint errors

### PR DESCRIPTION
**Pull request by digitalinfinity**
Mostly mechanical source transformations
Ran NAPI related c++ files through clang-format
Fixed remaining errors by hand

Fixes: #126

### CODE REVIEW

**Comment by digitalinfinity** [comment]
Note: this PR builds on top of #139 which is why there are two commits
Please review only 693ccff

cc @mhdawson @jasongin @gabrielschulhof

**Comment by digitalinfinity** [comment]
@sampsongao this will probably cause conflicts with your fix for #136 (and it might affect others too since its pretty pervasive) - I'll try to land this asap to minimize disruption

File: src/node_jsvmapi.h [hunk 1]
```diff
@@ -94,60 +81,42 @@ NODE_EXTERN napi_status napi_create_object(napi_env e, napi_value* result);
...
                                            napi_value* result);
-NODE_EXTERN napi_status napi_create_string_utf8(napi_env e,
-                                                const char* s,
-                                                int length,
-                                                napi_value* result);
-NODE_EXTERN napi_status napi_create_string_utf16(napi_env e,
-                                                 const char16_t* s,
+NODE_EXTERN napi_status napi_create_string_utf8(napi_env e, const char* s,
+                                                int length, napi_value* result);
+NODE_EXTERN napi_status napi_create_string_utf16(napi_env e, const char16_t* s,
```
Comment (mhdawson) [inline src/node_jsvmapi.h]: For these cases, when I looked at the existing code (I checked about 15-20 files)  in core, when there were multiple lines for the parameters the examples I found had one line per parameter. That's why I used that format in the changes I made to this file.  I think we should stick to that here and in the other files.

**Review by mhdawson — CHANGES_REQUESTED** [review]

File: src/node_jsvmapi.h (hunk 1 above)
Comment (digitalinfinity) [reply src/node_jsvmapi.h]: Ok- I reverted those changes- please take a look at the new commit

**Comment by digitalinfinity** [comment]
Replaced the previous commit with a new one- this time around formatted the sources with clang-format using the following commandline:
`clang-format -style="{BasedOnStyle: Google, BinPackArguments: false, BinPackParameters: false, PointerAlignment: Left, AllowAllParametersOfDeclarationOnNextLine: false}"`

**Comment by digitalinfinity** [comment]
Resolved merge conflicts and also fixed a couple warnings in NAPI code. @mhdawson please take a look when you get a chance.

**Review by mhdawson — APPROVED** [review]
LGTM, 
CI run here: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/16/

**Comment by mhdawson** [comment]
Running locally I see this failure:

gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make[2]: Entering directory `/home/mhdawson/testlint/abi-stable-node/test/addons-abi/test_buffer/build'
  CXX(target) Release/obj.target/test_buffer/test_buffer.o
../test_buffer.cc: In function âvoid staticBuffer(napi_env, napi_callback_info)â:
../test_buffer.cc:110:35: error: âreintepret_castâ was not declared in this scope
                                   reintepret_cast<char *>(theText),
                                   ^
../test_buffer.cc:15:7: note: in definition of macro âNAPI_CALLâ
   if (theCall != napi_ok) {                                               \
       ^
../test_buffer.cc:110:51: error: expected primary-expression before âcharâ
                                   reintepret_cast<char *>(theText),
                                                   ^
../test_buffer.cc:15:7: note: in definition of macro âNAPI_CALLâ
   if (theCall != napi_ok) {                                               \
       ^

on my local linux, I guess we'll see if that recreates in the CI as well


looks like a typo in  reintepret_cast, we should see CI fail as well.

Did you run '''make test''' after the changes ?

**Comment by digitalinfinity** [comment]
Hmm, I did run make test but I think I had something cached because I ran it again and it didn't hit that error either 😞 - fixed that typo and updated my PR. However, I do now see unit test failures in addons-abi related to napi_module_register being undefined. I suspect it's related to 14318a8 - I can take a look at it later tonight but @boingoing @jasongin you guys might want to take a look if you get a chance

**Comment by mhdawson** [comment]
I had done a run on master here: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/19/ and it ran ok, so not sure what you might be seeing:

New CI run on your updated PR: https://ci.nodejs.org/job/node-test-commit-linux-abi/21/

**Comment by mhdawson** [comment]
This is the error I see in the CI on your latest:

```bash
../test_buffer.cc: In function 'void staticBuffer(napi_env, napi_callback_info)':
../test_buffer.cc:106:89: error: reinterpret_cast from type 'const char*' to type 'char*' casts away qualifiers
   NAPI_CALL(
                                                                                         ^
../test_buffer.cc: In function 'void Init(napi_env, napi_value, napi_value)':
../test_buffer.cc:132:37: warning: missing initializer for member 'napi_property_descriptor::getter' [-Wmissing-field-initializers]
       {"staticBuffer", staticBuffer}};
                                     **^**
```

**Comment by mhdawson** [comment]
Maybe you can do a make clean or re-checkout to validate without having to wait for ci run ?

**Comment by mhdawson** [comment]
What is current checked in definitely builds/passes the tests in the CI: 

https://ci.nodejs.org/job/node-test-commit-linux-abi/nodes=ubuntu1404-64/23/console

So failures I think must be in your branch as show above.

**Review by boingoing — APPROVED** [review]
LGTM 👍

**Comment by digitalinfinity** [comment]
Thanks @mhdawson - apologies for the confusion here, I realized what was going on- make clean does not delete the test directories for me for some reason (it does surprisingly little on my Linux vm actually), and make wasn't picking up the changes in that folder and rebuilding it either. Doing a git clean worked and I was able to repro your error- fixed and also rebased my change on top of @sampsongao's change. I've verified locally that this clean build builds fine and make test also works (I verified that the addons-abi folder was built- there are warnings from napi_property_descriptor though as you point out above- I don't know if those need to be fixed or not, but if needed I can fix it in a separate commit).

CI is [here](https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/nodes=ubuntu1404-64/28/consoleText) - it looks good to my untrained eye. @mhdawson please let me know what you think.

**Comment by digitalinfinity** [comment]
Bah- @boingoing beat me to the punch- I'll rebase again and merge on top of his.

**Comment by digitalinfinity** [comment]
Rebased and pushed. @mhdawson please take a look.
CI: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/34/

**Review by mhdawson — APPROVED** [review]
LGTM.  If you need to rebase again, feel free to land after rebase, green CI run.

**Comment by digitalinfinity** [comment]
Rebased again and CI looks green to me: https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-commit-linux-abi/37/

Going to land this one now- thanks for the review @mhdawson

**Pull request merged**